kubernetes-deployment.md

```diff
@@ -0,0 +1,60 @@
+# Kubernetes Deployment
+
+### Prerequisites
+Docker Community Edition: Tested on docker version ```18.09.6```
+
+### Steps
+Set up Kubernetes cluster on your system by running the following script:
+```
+bash k8s-deployment/dind_k8s_cluster.sh up
+```  
+1. The script downloads kubectl and sets it in ```/usr/local/bin``` PATH environment variable.  
+**NOTE:**  
+i. It downloads kubectl version ```v1.13.0```, if you already have some other version of kubectl downloaded keep a backup of it.  
+ii. Ensure ```/usr/local/bin``` is set in your PATH environment variable as this is where the script places your kubectl.  
+2. The script sets up Kubernetes cluster (Version v1.13) based on DIND(Docker in Docker). An alternate version of Kubernetes can be set up by following steps given in this [link](https://github.com/kubernetes-sigs/kubeadm-dind-cluster).   
+3. Upon successful execution of the script a unique link to access the kubernetes-dashboard is generated.  
+![kubernetes-dashboard](images/dashboard.png)
+
+## OMS and KernelServer deployment in Kubernetes
+### Prerequisites
+1. Set credentials ```repo.username``` and ```repo.password``` in ```Amino.Run-Demos/gradle.properties``` file for access to the [dockerhub](https://hub.docker.com) repo.
+2. In the same file, change the port in the ```baseURL``` and replace it with the proxy port of kubernetes (Proxy port varies as per local kubernetes deployment).  
+![kubernetes-proxy-port](images/dashboard.png)
+3. Ensure generation of fatJar for your application.  
+    ```
+    bash gradlew build
+    ```
+
+### Steps
+Deploy OMS and kernelServer by executing the following command:
+```
+bash gradlew <appName>:subprojects:deployKernelServer
+```
+1. Upon execution of the above task a Dockerfile gets generated in respective ```subproject``` folder of application.
+2. The same Dockerfile is further used to build an image. This image contains the fatJar file and is used to run the OMS, kernelServer and the application.
+3. The image is then pushed to the [dockerhub](https://hub.docker.com) using the credentials set in the gradle.properties file: 
+    ```
+    repo.username=xxxx
+    repo.password=xxxx
+    ```
+4. Further deployment yaml files for OMS and kernelServer is generated using ```oms.template.yml``` and ```kernelserver.template.yml``` in respective applications.  
```

Comment:
IMHO, "in respective applications" can be changes to "in ```subproject``` folder of respective application with oms.yml and kernelserver.yml name"

Reply:
Added the above.